### PR TITLE
Show code for blocks without description #96 MVP

### PR DESCRIPTION
**Pull request by hexmind**

### CODE REVIEW

File: src/main/groovy/com/athaydes/spockframework/report/internal/HtmlReportCreator.groovy [hunk 1]
```diff
@@ -32,6 +32,9 @@ class HtmlReportCreator extends AbstractHtmlCreator<SpecData>
 
     HtmlReportCreator() {
         reportAggregator = defaultAggregator
+        if(showCodeBlocks){
+            log.warn('"showCodeBlocks" is not implemented in HtmlReportCreator. Use TemplateReportCreator.')
```
Comment (renatoathaydes) [inline src/main/groovy/com/athaydes/spockframework/report/internal/HtmlReportCreator.groovy]: Oh no! no support for HTML?

I guess I can do that myself later.

File: src/main/groovy/com/athaydes/spockframework/report/template/TemplateReportCreator.groovy [hunk 2]
```diff
@@ -112,19 +116,26 @@ class TemplateReportCreator implements IReportCreator {
...
-            else
+            } else {
                 [ : ]
+            }
         }.findAll { !it.empty }.flatten()
     }
 
+    private List<String> getBlockTexts(FeatureInfo feature, BlockInfo block) {
+        List<String> blockCode = showCodeBlocks ? codeReader.getLines(feature, block) : []
+        return blockCode ? blockCode : block.texts
```
Comment (renatoathaydes) [inline src/main/groovy/com/athaydes/spockframework/report/template/TemplateReportCreator.groovy]: `blockCode ?: block.texts`

File: src/main/groovy/com/athaydes/spockframework/report/vivid/SpecSourceCode.groovy [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.athaydes.spockframework.report.vivid
+
+import groovy.transform.ToString
+import org.codehaus.groovy.ast.MethodNode
+import org.spockframework.runtime.model.BlockKind
+
+@ToString
+class SpecSourceCode {
```
Comment (renatoathaydes) [inline src/main/groovy/com/athaydes/spockframework/report/vivid/SpecSourceCode.groovy]: I see you're using types everywhere, so can you try to make this use `@CompileStatic`?

File: src/main/groovy/com/athaydes/spockframework/report/vivid/VividAstInspector.groovy [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+import org.spockframework.util.inspector.AstInspector
+import org.spockframework.util.inspector.AstInspectorException
+import org.spockframework.util.inspector.Inspect
+
+import java.security.CodeSource
+
+/**
+ * Based on org.spockframework.util.inspector.AstInspector by Peter Niederwieser
+ */
+public class VividAstInspector extends AstInspector{
```
Comment (renatoathaydes) [inline src/main/groovy/com/athaydes/spockframework/report/vivid/VividAstInspector.groovy]: What's up with `Vivid`?

**Comment by renatoathaydes** [comment]
Wow, this is a nice addition!!!

Great PR. The code looks clean and well tested... I understand that you focussed on the template report (I guess that's what you're using) but to be complete, we should add support for the HTML reports as well... given what you've done, I don't mind if you leave that for me :) After all your work, looks like doing that will be pretty simple.

**Comment by renatoathaydes** [comment]
Ah, looks like there's conflicts... I guess you didn't notice I have some changes lined up for the next release... hope you can fix easily.

File: src/main/groovy/com/athaydes/spockframework/report/internal/HtmlReportCreator.groovy (hunk 1 above)
Comment (hexmind) [inline src/main/groovy/com/athaydes/spockframework/report/internal/HtmlReportCreator.groovy]: It is first, minimal implementation. That's the reason.

File: src/main/groovy/com/athaydes/spockframework/report/vivid/VividAstInspector.groovy (hunk 4 above)
Comment (hexmind) [inline src/main/groovy/com/athaydes/spockframework/report/vivid/VividAstInspector.groovy]: This is codename for this feature. Initially it was "descriptive" for descriptive and meaningful code but vivid is shorter one.

**Comment by hexmind** [comment]
Now it is rebased with "next" branch without conflicts and with your comments applied. You can continue with html reports.
Later when it will be well tested on real cases (markdown/html) we can rename PROP_SHOW_CODE_BLOCKS and change default value.

**Comment by renatoathaydes** [comment]
I don't have a lot of time right now, so I will make a release with my changes (improved report looks) and your additions... will try to add html support for this feature in a later release.

**Comment by renatoathaydes** [comment]
Unfortunately I can't release it yet. There's a bug in that nested classes are not found by the AST visitor...

**Comment by renatoathaydes** [comment]
Actually, it's a Spec defined in a file with different name. Eg.

```groovy
// file name = MySpec.groovy

class MySpec extends Specification {
    def "2 + 2 is 4"() {
        expect:
            2 + 2 == 4
}

class MyOtherSpec extends Specification {
    def "1 + 1 is 2"() {
        expect:
            1 + 1 == 2
}
```

**Pull request merged**